src/Escrow.sol

```diff
@@ -220,6 +223,32 @@ contract Escrow {
         emit Deposit(msg.sender, receiver, amount);
     }
 
+    /**
+     * @dev Deposits escrow for multiple receivers.
+     * @param receivers Array of addresses of the receivers. 
+     * @param amounts Array of amounts of escrow to deposit.
+     * @notice The escrow must be approved for transfer by the sender.
+     * @notice REVERT: this function will revert if the escrow transfer fails.
+     * @notice REVERT: if the length of the receivers and amounts arrays do not match.
+     */
+    function depositMany(address[] calldata receivers, uint256[] calldata amounts) external {
+        if (receivers.length != amounts.length) {
+            revert InputsLengthMismatch();
+        }
+
+        uint256 totalAmount = 0;
+        for (uint i = 0; i < receivers.length; i++) {
+            address receiver = receivers[i];
+            uint256 amount = amounts[i];
+
+            totalAmount += amount;
+            escrowAccounts[msg.sender][receiver].balance += amount;
+            emit Deposit(msg.sender, receiver, amount);
+        }
+
+        escrowToken.safeTransferFrom(msg.sender, address(this), totalAmount);
```

Comment:
Is the reason this doesn't just call `deposit()` in the loop to avoid calling `safeTransferFrom()` multiple times/ gas savings? Otherwise, it would be more maintainable if we called `deposit()` directly instead of duplicating code.

Reply:
Yes exactly. We could create a `_deposit()` function that updates state and made the transfer but we would be making N transfers instead of 1. If you have 30/50 receivers in one call that's a lot of wasted gas. I agree that it would be more maintainable though.